src/dmd/globals.d

```diff
@@ -289,8 +289,9 @@ extern (C++) struct Param
     const(char)[] moduleDepsFile;        // filename for deps output
     OutBuffer* moduleDeps;              // contents to be written to deps file
 
+    bool emitMakeDeps;                   // whether to emit makedeps
     const(char)[] makeDepsFile;          // filename for makedeps output
-    OutBuffer* makeDeps;                 // contents to be written to makedeps file
+    Array!(const(char)*) makeDeps;      // dependencies for makedeps
```

Comment:
```suggestion
    Array!(const(char)[]) makeDeps;      // dependencies for makedeps
```

Reply:
Maybe it should be moved to a dmd-specific parameter then? (though still uncommitted #11762)

Reply:
How is this implemented for GDC @ibuclaw? If the GCC codebase handles this for you, perhaps it could still be useful for LDC. @kinke what do you think?

Reply:
AFAIK LDC do not have this feature, and will have it "for free" once this version of frontend will be merged.

Reply:
`Array!(const(char)[])` do not compile.
```
src/dmd/root/array.d(52): Error: Internal Compiler Error: type const(char)[] cannot be mapped to C++
```